src/vs/workbench/browser/parts/notifications/notificationsTelemetry.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { IWorkbenchContribution, IWorkbenchContributionsRegistry, Extensions } from 'vs/workbench/common/contributions';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { hash } from 'vs/base/common/hash';
+
+interface NotificationMetrics {
+	id: number;
+	source?: string;
+}
+
+type NotificationMetricsClassification = {
+	id: { classification: 'SystemMetaData', purpose: 'FeatureInsight' };
+	source?: { classification: 'SystemMetaData', purpose: 'FeatureInsight' }
+};
+
+class NotificationsTelemetryContribution extends Disposable implements IWorkbenchContribution {
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@INotificationService notificationService: INotificationService
+	) {
+		super();
+		this._register(notificationService.onDidAddNotification(notification => {
+			if (!notification.silent) {
+				telemetryService.publicLog2<NotificationMetrics, NotificationMetricsClassification>('notification:show', {
+					id: hash(notification.message.toString()),
+					source: notification.source
+				});
+			}
+		}));
+
+		this._register(notificationService.onDidRemoveNotification(notification => {
+			if (!notification.silent) {
+				telemetryService.publicLog2<NotificationMetrics, NotificationMetricsClassification>('notification:close', {
+					id: hash(notification.message.toString()),
+					source: notification.source
+				});
+			}
+		}));
+	}
+}
+
+export function registerNotificationTelemetry() {
```

Comment:
I do not think this indirection through a `IWorkbenchContribution` is needed because the `workbench.ts` can directly instantiate this thing, no? From the workbench I would have just expected a call like `this._register(instantiationService.createInstance(NotificationsTelemetry...`

Reply:
@bpasero yes I can change that, just for style I went with the current approach.